File: Array/2530.Maximal_Score_After_Applying_K_Operations/C++/2530.Maximal_Score_After_Applying_K_Operations.cpp
class Solution {
public:
    long long maxKelements(vector<int> & nums, int k) 
    {
        //================================================//
        // Input type:                                    //
        //   - vector<int> ref. (nums vector)             //
        //   - int (k-th operations)                      //
        // Return type:                                   //
        //   - long long (result maximun summary counter) //
        //================================================//

        /*Initialize*/
        ///// Length of nums vector /////
        int len_nums = nums.size();

        ///// Record heap, priority-queue /////
        priority_queue<int> record_pq(begin(nums), end(nums));

        ///// Record value /////
        int record_val;

        ///// Result maximun summary counter /////
        long long res_max_cnter = 0;


        /*Heap, Priority-queue based loop traversal*/
        for (int k_idx = 0; (k_idx < k); (k_idx++)) //K-th operations
        {
            record_val = record_pq.top(); //Record value
            record_pq.pop(); //Keep updating, popping

            res_max_cnter += record_val; //Keep updating, accumulating

            ///// Check if the current indexed-value is divisable by 3 or not /////
            if (! (record_val % 3)) { record_pq.push((record_val / 3)); } //Keep updating, recording
            else { record_pq.push(((record_val / 3) + 1)); } //Keep updating, recording

        } //K-th operations

        return res_max_cnter;
    }
};

File: Array/2530.Maximal_Score_After_Applying_K_Operations/Python/2530.Maximal_Score_After_Applying_K_Operations.py
class Solution(object):
    def maxKelements(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        #=======================================#
        # Priority-queue based traversal method #
        #=======================================#

        ############
        #Initialize
        ##### Length of nums array #####
        len_nums = len(nums)

        ##### Updated nums array #####
        for nums_idx in range(len_nums):
            nums[nums_idx] *= (-1)

        heapify(nums)

        ##### Result maximun summary counter #####
        res_max_cnter = 0


        #####################################
        #Priority-queue based loop traversal
        for k_idx in range(k):
            curr_min_val = heappop(nums) #Current minimun value

            res_max_cnter -= curr_min_val #Keep updating/accumulating

            heappush(nums, int(floor(curr_min_val / 3))) #Keep updating/recording

        return res_max_cnter
